blank-canvas-blocks/assets/navigation.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * File navigation.js.
+ *
+ * Required to open the mobile navigation.
+ */
+(function () {
+
+	function addMobileMenuButton( navMenu ) {
+
+		const menuContainer = navMenu.querySelector( '.wp-block-navigation__container' );
+		const openButton = document.createElement( 'a' );
+		const openButtonLabel = getComputedStyle(menuContainer).getPropertyValue( '--wp--custom--navigation--mobile--menu--label' );
+		openButton.setAttribute( 'href', '#' );
+		openButton.classList.add( 'wp-block-navigation__mobile-menu-open-button' );
+		openButton.innerText = ( openButtonLabel );
+		openButton.addEventListener( 'click', () => {
+			navMenu.classList.add( 'show' );
+			menuContainer.scrollTop = 0;
+		});
+
+		menuContainer.addEventListener( 'click', (mouseEvent) => {
+			if ( mouseEvent.target === menuContainer ) {
+				navMenu.classList.remove( 'show' );
+			}
+		});
+
+		navMenu.appendChild( openButton );
+	}
+
+	window.addEventListener( 'load', () => {
+		Array.from( document.querySelectorAll( 'header .wp-block-navigation' ) )
+			.forEach( addMobileMenuButton );
```

Comment:
By this logic — there could be multiple nav blocks in the header, and each one would receive its own separate mobile menu?

Reply:
Yes, that's correct.  Initially I wrote that leveraging an "opt in class" rather than using the header.

Either way though that maybe should change to just `.querySelector` so we just execute this behavior on the FIRST menu that is in the header.  Which might not always be the menu that we want but I'm not sure what else would be best...

Reply:
Refactored to use `querySelector` instead.  If/when the multiple menus comes up we'll refactor as needed then.